### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Recillah-Khamala**
Hello code reviewer

In this project, my partner and I implemented all the project requirements as stipulated below 
- We followed the layout of the wireframes provided.
- Login page and registration page:
  - We used devise gem for this section.
- Food list:
  - It only displays a list of food added by the logged-in user as in the wireframe.
  - Has a button that leads to a form that allows users to add new food.
- Recipes list:
  - It only displays a list of recipes created by the logged-in user as in the wireframe.
  - Clicking the title lead to recipe details.
  - If the user is the recipe owner, they can delete it.
- Public recipe list:
  - It displays a list of all public recipes ordered by the newest as in the wireframe.
  - Clicking on it lead to recipe details.
- Recipe details:
  - If the recipe is public or the user is the recipe owner, it displays the recipe details as in the wireframe.
  - If the user is the recipe owner, it has a button that leads to the form that allows the user to add new food.
- General shopping list view
  - It shows the list of food missing for all recipes of the logged-in user.
  - It counts the total number of food items and the total price of the missing food.
- Used bullet gem to make sure there are no N+1 queries happening.
- Created a navigation menu that allows users to open all of the pages you created.

### CODE REVIEW

File: app/views/recipes/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      </div>
+    <% end %>
+  </div>
+
+  <div class="mt-8">
+    <% if @recipes.empty? %>
+      <p>There are no recipes.</p>
+    <% end %>
+  </div>
+</div>
```
Comment (VICTOR-CHUKWU) [inline app/views/recipes/index.html.erb]: - [ ] your recipe index page is not following the design on the wireframe, kindly fix that
your design :point_down: 
Screenshot from 2023-02-18 17-03-50
wireframe :point_down: 
Screenshot from 2023-02-18 17-06-53
kindly move your description the right hand side of the cards

File: app/views/recipes/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+	</div>
+<h2 class="m-4 text-indigo-black font-bold">The Recipe details go here: </h2>
+<div class="mt-2">
+  <% if !current_user.nil? && @recipe.recipe_foods.empty? %>
+    <h1 class="text-2xl font-bold text-center text-indigo-600">
+      "Not Found !!!"
+    </h1>
+  <% else %>
+    <table class="m-auto border border-slate-500">
+      <thead>
```
Comment (VICTOR-CHUKWU) [inline app/views/recipes/show.html.erb]: - [ ] nice job so far, but your recipe show page does not look like the design on the wireframe, kindly make it look similar to the wireframe
your design :point_down: 
Screenshot from 2023-02-18 17-12-31
wireframe :point_down: 
Screenshot from 2023-02-18 17-24-13

File: app/views/recipes/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	</div>
+	<div class="text-indigo-black w-full m-2 flex justify-start items-center">
+	<h3 class="font-semibold">Cooking time: </h3><h6 class="ml-2"> <%= @recipe.cooking_time %></h6>
+	</div>
+	<div class="w-full m-2 flex justify-start items-center">
+	<h3 class="text-indigo-black font-semibold">Description: </h3><p class="ml-2 w-full"> <%= @recipe.description %></p>
+	</div>
+	<% if !current_user.nil? && current_user.id == @recipe.user_id %>
+		<div class="w-full my-4 flex justify-between items-center">
+			<%= link_to "Generate shopping list", recipe_new_shopping_list_path(@recipe.id), class: "bg-red-500 px-4 py-2 font-semibold text-white inline-flex items-center space-x-2 rounded"%>
```
Comment (VICTOR-CHUKWU) [inline app/views/recipes/show.html.erb]: - [ ] onclick of this generate shopping should take me to shopping list page, but now it throws an error if the recipe has ingredients, kindly fix this
Screenshot from 2023-02-18 17-25-45

File: app/views/recipes/public_recipes.html.erb [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+        <p class="text-gray-600">Total food items: <%= recipe.foods.length %></p>
+        <p class="text-gray-600">
+          Total cost: <%= recipe.foods.reduce(0) do |sum, food| %>
+            <% sum + (food.quantity * food.price)  %>
+          <% end %>
+        </p>
+      </div>
+    </div>
+  <% end %>
+</div>
```
Comment (VICTOR-CHUKWU) [inline app/views/recipes/public_recipes.html.erb]: - [ ] also your public recipe page does not match the design in the wireframe, kindly fix this
Screenshot from 2023-02-18 17-32-00

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @Recillah-Khamala  and @Doheera-kosi  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- All linters passed


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by ichala — APPROVED** [review]
Hi Student  ,
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:  
 
68747470733a2f2f6d65646961322e67697068792e636f6d2f6d656469612f6c3366516333614356304871707a5763552f67697068792e676966

# To Highlight 💪🏻

- Changes Implemented ✔️ 


 Good job implementing the requirements  :+1: Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

# Optional Suggestions  :

- N/A


**********
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag  ichala in your question so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**